### PR TITLE
[Comment][Relay] Add comment APIs 

### PR DESCRIPTION
**Pull request by Lisooo790926**
Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary
Add comments related apis 
- Get comments API 
- Leave Comment API
- Comments test

## Linked Issue
#137
#127 

## Details
1. Review the [spec of comment](https://docs.google.com/document/d/1bf7vHbuMBchrUw_dfKDMFLQVXILqEvKsnL7dJkUPlKw/edit) 
1. get comment api 
2. leave comment api 
3. Move common method to EpochKeyService to reuse in post and comment

## Impacted Areas
Comment related feature

## Tests
UnitTest 

## Possible Impacts
Comment related feature

## Verification Steps
1. Leave comment with login or no login user
2. Leave comment with empty content
3. Leave comment with wrong epoch key proof
4. Fetch comment with postId or epochKey 

## Todo
-   [x] Review the requirement doc again and test above apis
-   [x] Draft implementation for `function`
-   [x] Unit tests for `function`

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities
-   [x] Run `yarn lint:fix`

### CODE REVIEW

File: packages/relay/src/routes/post.ts [hunk 1]
```diff
@@ -20,167 +15,54 @@ export default (
...
 
     app.get(
         '/api/post/:id',
         errorHandler(async (req, res, next) => {
-            await fetchSinglePost(req, res, db)
-        })
-    )
-}
-
-async function fetchPosts(req, res, db: DB) {
```
Comment (Lisooo790926) [inline packages/relay/src/routes/post.ts:42]: Move all related logic to `PostService`

File: packages/relay/src/singletons/schema.ts [hunk 2]
```diff
@@ -77,6 +77,25 @@ const _schema = [
...
+            ['commentId', 'String', { optional: true }],
+            ['postId', 'String'],
+            ['cid', 'String', { optional: true }],
+            ['transactionHash', 'String'],
+            ['content', 'String'],
+            ['epoch', 'Int'],
+            ['epochKey', 'String'],
+            ['status', 'Int'],
+        ],
+    },
```
[the remark this answers is not in the document]
Comment (Lisooo790926) [reply packages/relay/src/singletons/schema.ts:98]: got it thanks~

[the remark this answers is not in the document]
Comment (chentihe) [reply packages/relay/src/singletons/schema.ts:98]: Duplicate schema, not remove yet

File: packages/relay/test/counter.test.ts [hunk 3]
```diff
@@ -111,11 +110,16 @@ describe('GET /counter', function () {
...
     it('should delete the EpochKeyAction table after the epoch ended', async function () {
-        const epoch = sync.calcCurrentEpoch()
+        const epoch = await sync.loadCurrentEpoch()
         const epochRemainingTime = sync.calcEpochRemainingTime()
+
+        await ethers.provider.send('evm_increaseTime', [
+            epochRemainingTime + 100000,
+        ])
+
         // add 10s to make sure this epoch ended
```
Comment (chentihe) [inline packages/relay/test/counter.test.ts:120]: modify this comment and move before
```js
await ethers.provider.send('evm_increaseTime', [
            epochRemainingTime + 100000,
        ])
```
you used ethers to fast forward block, then you can remove unnecessary package
`@nomicfoundation/hardhat-network-helpers`

File: packages/relay/src/routes/post.ts [hunk 4]
```diff
@@ -20,167 +15,54 @@ export default (
...
-
-            // store content into helia ipfs node with json plain
-            const { json } = await eval("import('@helia/json')")
-            const heliaJson = json(helia)
-            const IPFSContent = {
-                content: content,
+            const id = req.params.id
+            if (!id) {
+                console.log('id is undefined')
+                res.status(400).json({ error: 'id is undefined' })
```
Comment (chentihe) [inline packages/relay/src/routes/post.ts:56]: should add return to finish this request
or should throw error like comment api

File: packages/relay/src/routes/post.ts [hunk 5]
```diff
@@ -20,167 +15,54 @@ export default (
...
-        }
-
-        const post = await db.findOne('Post', {
-            where: {
-                _id: id,
-            },
+            const post = await postService.fetchSinglePost(id, db, undefined)
+            if (!post) {
+                console.log(`post is not found: ${id}`)
+                res.status(404).json({ error: `post is not found: ${id}` })
```
Comment (chentihe) [inline packages/relay/src/routes/post.ts:62]: should add return to finish this request
or should throw error like comment api

File: packages/relay/src/singletons/schema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lisooo790926) [reply packages/relay/src/singletons/schema.ts:98]: Sorry forgot to push this change

File: packages/relay/src/routes/post.ts (hunk 5 above)
Comment (Lisooo790926) [reply packages/relay/src/routes/post.ts:62]: I think this one is find, since it is the end of method

**Review by chentihe — APPROVED** [review]

**Pull request merged**